sale_multi_operating_unit/models/sale_order.py

```diff
@@ -0,0 +1,124 @@
+            self.name = sale_id.name + ' - ' + self.operating_unit_id.code
+
+    @api.onchange('state')
+    def _onchange_state(self):
+        lead_obj = self.env['crm.lead']
+        if self.state and self.state == 'sent' and self.sale_id:
+            lead_id = lead_obj.search([('partner_id', '=',
+                                        self.sale_id.partner_id.id),
+                                       ('stage_id', '=', 4)])
+            if not lead_id:
+                lead_id = lead_obj. \
+                    create({
+                           'partner_id': self.sale_id.partner_id.id,
+                           'type': 'lead',
+                           'user_id': self._uid,
+                           'name': self.sale_id.name + ' - ' +
+                           self.operating_unit_id.code
+                           })
+                self.lead_id = lead_id.id
+
+
+class SaleOrderQuoteLine(models.Model):
+    _name = 'sale.order.quote.line'
+
+    name = fields.Char(string='Description', required=True)
+    quote_id = fields.Many2one(
+        comodel_name='sale.order.quote',
+        string='Internal Quote'
+    )
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product'
+    )
+    qty = fields.Float(string="Quantity")
+    uom_id = fields.Many2one(
+        comodel_name='uom.uom',
+        string='Unit of Measure'
+    )
+    price_unit = fields.Float(string="Unit Price")
+    subtotal = fields.Float(string="Subtotal")
+
+    @api.multi
+    @api.onchange('product_id')
+    def product_id_change(self):
+        if not self.product_id:
+            return {'domain': {'product_uom': []}}
+
+        vals = {}
+        domain = {'product_uom': [('category_id', '=',
+                                   self.product_id.uom_id.category_id.id)]}
+        if not self.uom_id or (self.product_id.uom_id.id != self.uom_id.id):
+            vals['uom_id'] = self.product_id.uom_id
+            vals['qty'] = self.qty or 1.0
+
+        result = {'domain': domain}
+
+        vals['price_unit'] = self.product_id.lst_price
+        name = self.product_id.get_product_multiline_description_sale()
+        vals.update(name=name)
+        self.update(vals)
+        return result
+
+    @api.multi
+    @api.onchange('qty')
+    def _onchange_qty(self):
+        if self.qty:
+            self.subtotal = self.price_unit * self.qty
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    quote_ids = fields.One2many(
+        'sale.order.quote', 'sale_id', string='Internal Quotes')
+
+    @api.constrains('quote_ids')
+    def _operating_unit_id(self):
+        operating_unit = [str(quote.operating_unit_id.id)
+                          for quote in self.quote_ids]
+        result = len(operating_unit) != len(set(operating_unit))
+        if result:
+            raise ValidationError(_(
+                """The Operating Unit must be unique per sale order!"""))
```

Comment:
@Dangar-Serpentcs Can you please split to have one file per model?

Ref:
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#15file-naming